extension/js/common/helpers.ts

```diff
@@ -3,8 +3,119 @@
 'use strict';
 
 import { AcctStore } from './platform/store/acct-store.js';
-
+import { PassphraseOptions } from '../../chrome/settings/setup.js';
+import { Buf } from './core/buf.js';
+import { Key, KeyInfoWithIdentity, KeyUtil } from './core/crypto/key.js';
+import { ClientConfiguration } from './client-configuration.js';
+import { ContactStore } from './platform/store/contact-store.js';
+import { KeyStore } from './platform/store/key-store.js';
+import { PassphraseStore } from './platform/store/passphrase-store.js';
+import { Bm } from './browser/browser-msg.js';
 export const isFesUsed = async (acctEmail: string) => {
   const { fesUrl } = await AcctStore.get(acctEmail, ['fesUrl']);
   return Boolean(fesUrl);
 };
+
+export const saveKeysAndPassPhrase = async (acctEmail: string, prvs: Key[], options?: PassphraseOptions) => {
+  const clientConfiguration = await ClientConfiguration.newInstance(acctEmail);
+  for (const prv of prvs) {
+    await KeyStore.add(acctEmail, prv);
+    if (options !== undefined) {
+      await PassphraseStore.set((options.passphrase_save && !clientConfiguration.forbidStoringPassPhrase()) ? 'local' : 'session',
+        acctEmail, { longid: KeyUtil.getPrimaryLongid(prv) }, options.passphrase);
+    }
+  }
+  const { sendAs, full_name: name } = await AcctStore.get(acctEmail, ['sendAs', 'full_name']);
+  const myOwnEmailsAddrs: string[] = [acctEmail].concat(Object.keys(sendAs!));
+  for (const email of myOwnEmailsAddrs) {
+    if (options !== undefined) {
+      // first run, update name
+      // todo: refactor?
+      await ContactStore.update(undefined, email, { name });
+    }
```

Comment:
> Typically we'd want to test for `typeof options !== 'undefined'` because javascript is odd.

Hmm... I googled a bit about this and didn't find any good reason for the `typeof` comparison, especially in Typescript, so I started to use ` === undefined` comparison in all my projects.
https://stackoverflow.com/questions/4725603/variable-undefined-vs-typeof-variable-undefined

Reply:
I see. I think I have been doing `typeof` because it's somewhat safer, if you happen to be testing global variable. It's true we don't do that much, but anyway it appears I've standardized on a single way to do it - I usually favor consistency.

I suppose it doesn't matter much either way.